services/sidecar/rollup/derivation/derivation_block.go

```diff
@@ -48,41 +49,37 @@ func (e *DecodeTxBatchError) Error() string {
 
 // Decodes the input of `SequencerInbox.appendTxBatch` call
 func BlocksFromData(calldata []any) ([]DerivationBlock, error) {
-	if len(calldata) != 4 {
+	if len(calldata) != 1 {
 		return nil, &DecodeTxBatchError{fmt.Sprintf("invalid decoded array length %d", len(calldata))}
 	}
 	var (
-		contexts           = calldata[0].([]*big.Int)
-		txLengths          = calldata[1].([]*big.Int)
-		firstL2BlockNumber = calldata[2].(*big.Int)
-		// TODO: commented until multiple versions have been used
-		//txBatchVersion     = calldata[3].(*big.Int)
-		txBatch = calldata[4].([]byte)
+		txBatchData = calldata[0].([]byte)
 	)
-	if len(contexts)%2 != 0 {
-		return nil, &DecodeTxBatchError{fmt.Sprintf("invalid contexts length %d", len(contexts))}
+
+	if len(txBatchData) == 0 {
+		return nil, &DecodeTxBatchError{fmt.Sprintf("empty tx batch data")}
 	}
-	var (
-		batchOffset, numTxs uint64
-		blocks              = make([]DerivationBlock, 0, len(contexts)/2)
-		blockNumber         = firstL2BlockNumber.Uint64()
-	)
-	for i := 0; i < len(contexts); i += 2 {
+
+	txBatchVersion := txBatchData[0]
+	if txBatchVersion != TxBatchVersion() {
```

Comment:
can we dispatch to different version of decoders instead of asserting 1 version? For other version return unimplemented.

Reply:
added a switch statement to dispatch, right now the only decoder is `blocksFromV0Data`